MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,72 +1,121 @@
 import UIKit
 
-final class MovieQuizViewController: UIViewController {
-    // MARK: - Lifecycle
-    override func viewDidLoad() {
+final class MovieQuizViewController: UIViewController, QuestionFactoryDelegate {
+    func didReceiveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else {return}
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        DispatchQueue.main.async { [weak self] in
+            self?.show(quiz: viewModel)
+        }}
+    
+    override func viewDidLoad()
+    {
         super.viewDidLoad()
+        imageView.layer.masksToBounds = true
+        imageView.layer.cornerRadius = 20
+        alertPresenter = AlertPresenter(alertController: self)
+        questionFactory = QuestionFactory(delegate: self)
+        questionFactory?.requestNextQuestion()
+        statisticService = StatisticServiceImplementation()
+    }
+    
+    private var correctAnswers: Int = 0
+    private var currentQuestionIndex: Int = 0
+    private let questionsAmount: Int = 10
+    private var alertPresenter : AlertPresenter?
+    private var currentQuestion: QuizQuestion?
+    private var questionFactory: QuestionFactoryProtocol? = nil
+    private var statisticService: StatisticService?
+    
+    @IBOutlet private var imageView: UIImageView!
+    @IBOutlet private var textLabel: UILabel!
+    @IBOutlet private var counterLabel: UILabel!
+    
+    @IBOutlet private weak var noButton: UIButton!
+    @IBOutlet private weak var YesButton: UIButton!
+    
+    @IBAction private func yesButtonClicked(_ sender: Any)
+    {
+        guard let currentQuestion = currentQuestion else { return }
+        let givenAnswer = true
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+    }
+    @IBAction private func noButtonClicked(_ sender: Any)
+    {
+        guard let currentQuestion = currentQuestion else { return }
+        let givenAnswer = false
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+    }
+    
+    
+    
+    private func showNextQuestionOrResults()
+    {
```

Comment:
Кажется я неточно выразилась, открывающая должна оставаться на той же строке:
```suggestion
    private func showNextQuestionOrResults() {
```